packages/fuels-programs/src/contract.rs

```diff
@@ -732,7 +739,7 @@ pub fn method_hash<D: Tokenizable + Parameterize + Debug, T: Account>(
 
     let compute_custom_input_offset = should_compute_custom_input_offset(args);
 
-    let unresolved_bytes = ABIEncoder::encode(args)?;
+    let unresolved_bytes = ABIEncoder::default().encode(args)?;
```

Comment:
The `method_hash` function should have the `encoder_configuration` as an input so that the `ABIEncoder` can be customized.

The user should be able to set the `encoder_configuration` wih the `contract_instance` and then pass it to `methods` which uses `method_hash`. The final result could look like this:
```rust
contract_instance.with_encoder_configuration(encoder_config).methods().my_method().call()
```
The `contract_instance` should have a default `encoder_config` at initialization so that this is also possible:
```rust
contract_instance.methods().my_method().call()
```